Sources/swift-build-prebuilts/main.swift

```diff
@@ -0,0 +1,437 @@
+                }
+
+                // Zip it up
+                stageDir.cd()
+                let zipFile = stageDir / "\(swiftVersion)-\(library.name)-\(platform).zip"
+                let contentDirs = ["lib", "Modules"] + (library.cModules.isEmpty ? [] : ["include"])
+                try zipFile.compress(contentDirs)
+                repoDir.cd()
+
+                newArtifacts.append(.init(platform: platform, checksum: try zipFile.checksum()))
+
+                try libDir.remove()
+                try modulesDir.remove()
+                try includesDir.remove()
+            }
+
+            let newLibrary = Workspace.PrebuiltsManifest.Library(
+                name: library.name,
+                products: library.products,
+                cModules: library.cModules,
+                artifacts: newArtifacts
+            )
+            newLibraries.insert(newLibrary)
+
+            try shell("git reset --hard")
+        }
+
+        if let oldManifest = try await downloadManifest(version: version) {
+            // Add in elements from the old manifest we haven't generated
+            for library in oldManifest.libraries {
+                if var newLibrary = newLibraries[library.name] {
+                    var newArtifacts = IdentifiableSet<Workspace.PrebuiltsManifest.Library.Artifact>(newLibrary.artifacts)
+                    for oldArtifact in library.artifacts {
+                        if !newArtifacts.contains(id: oldArtifact.id) {
+                            newArtifacts.insert(oldArtifact)
+                        }
+                    }
+                    newLibrary.artifacts = .init(newArtifacts.values)
+                    newLibraries.insert(newLibrary)
+                } else {
+                    newLibraries.insert(library)
+                }
+            }
+        }
+        let newManifest = Workspace.PrebuiltsManifest(libraries: .init(newLibraries.values))
+
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let manifestData = try encoder.encode(newManifest)
+        let manifestFile = stageDir / "\(swiftVersion)-manifest.json"
+        try manifestFile.write(manifestData)
+    }
+}
+
+func canBuild(_ platform: Workspace.PrebuiltsManifest.Platform) -> Bool {
+    if dockeronly {
+        return platform.os == .linux
+    }
+    #if os(macOS)
+    if platform.os == .macos {
+        return true
+    }
+    #elseif os(Windows)
+    if platform.os == .windows {
+        return true
+    }
+    #endif
+    return docker && platform.os == .linux
+}
+
+func shell(_ command: String) throws {
+    let process = Process()
+#if os(Windows)
+    process.executableURL = URL(fileURLWithPath: "C:\\Windows\\System32\\cmd.exe")
+    process.arguments = ["/c", command]
+#else
+    process.executableURL = URL(fileURLWithPath: "/bin/bash")
+    process.arguments = ["-c", command]
+#endif
+    print("Running:", command)
+    try process.run()
+    process.waitUntilExit()
```

Comment:
I don't think we should get into the habit of saying that it's OK as something could easily come along later which makes this unsafe. Can you please use the Process instance from TSCBasic instead of using the Foundation one directly for consistency with the rest of the codebase? (and because it has an async-compatible waitUntilExit overload)

Reply:
Done. Still sad to need to avoid using Foundation because of it's limitations.